### PR TITLE
Windows: Inject `vcruntime.apinotes` automatically

### PR DESCRIPTION
**Pull request by stevapple**
Extract `vcruntime.apinotes` to inside Swift SDK, to eliminate custom setup.

Resolves SR-NNNN.

### CODE REVIEW

**Comment by stevapple** [comment]
This is the first step to:
- ensure a non-invasive setup against Windows / MSVC SDK;
- align Windows build behavior with other platforms.

File: CMakeLists.txt [hunk 1]
```diff
@@ -148,7 +148,7 @@ option(SWIFT_INCLUDE_DOCS
     TRUE)
 
 set(_swift_include_apinotes_default FALSE)
-if(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+if(CMAKE_SYSTEM_NAME STREQUAL "Darwin" OR CMAKE_SYSTEM_NAME STREQUAL "Windows")
   set(_swift_include_apinotes_default TRUE)
 endif()
```
Comment (compnerd) [inline CMakeLists.txt:154]: The description of the option here is misleading, it isn't about inclusion in the toolchain it seems (based on apple/swift-installer-scripts#79) but rather into the installed image.  Could you please update the comment?

**Comment by compnerd** [comment]
Please test with following PRs:
apple/swift-installer-scripts#79

@swift-ci please build toolchain Windows platform

**Comment by stevapple** [comment]
> how do you intend to have differences between SDKs show up if this is bundled into the toolchain? This really belongs in the SDK.

The `apinotes` is a part of SDK, not the toolchain IMO. It works like an overlay to an existing framework.

**Comment by compnerd** [comment]
Ah, okay, as long as it stays within the SDK and can be applied without injection, that seems reasonable to me.  I don't think that the support for that was there when I started on the port ~5 years ago.

File: CMakeLists.txt (hunk 1 above)
Comment (stevapple) [reply CMakeLists.txt:154]: `apinotes` are regarded to be optional overlays, so we can choose to include or not include it in our build.

Comment (compnerd) [reply CMakeLists.txt:154]: Sure; but that doesn't change the fact that the comment for the CMake GUI is misleading.

**Comment by stevapple** [comment]
After some investigation (on Darwin because that's where builtin `apinotes` injection is already a thing), I found things become tricky now.

`apinotes` belongs to Swift runtime: not SDK or toolchain in general. On Darwin, Swift runtime is a part of toolchain (because of ABI stability I guess); on other platforms, that belongs to the SDK.

**Comment by stevapple** [comment]
Eventually we should move this into the toolchain. Since MSVC itself is known to be stable, this makes no problem.
However we have to make it alongside the Swift runtime, so let it stay in SDK now.

**Comment by compnerd** [comment]
MSVC is exactly the opposite: guaranteed unstable.  This should most definitely not be part of the toolchain and it should never be part of the runtime, there is no reason to ever distribute this to the users.  This is purely an SDK component.

**Comment by stevapple** [comment]
API notes are irrelated to ABI. They used to be regarded as compiler's knowledge of specific API/variable from specific modules.

If `visualc` doesn't prove to be API-compatible, maybe we should end up using SDK overlay instead. This allows us to overwrite the symbol with concrete type and more explicit availability annotation.

**Comment by stevapple** [comment]
That's too far for this PR — our runtime itself is depending on Windows API interface now, so does the API note. We ought to distribute it in a standard way.

**Comment by stevapple** [comment]
Any update for this? @compnerd 

FYI, `apinotes` is built along with Swift runtime libraries (the `swiftmodule`s), which is now considered to be part of SDK on Windows.

**Comment by stevapple** [comment]
Try to get this in as part of eliminating manual injections.

**Pull request closed without merging**